### PR TITLE
Add initial "hostile contract" and unit test calling it

### PR DESCRIPTION
**Pull request by jayz22**
### What

To merge https://github.com/stellar/rs-soroban-env/pull/466

### Why

[TODO: Why this change is being made. Include any context required to understand the why.]

### Known limitations

[TODO or N/A]

### CODE REVIEW

**Comment by jayz22** [comment]
Can I get a merge on this?

File: soroban-env-common/src/meta.rs [hunk 1]
```diff
@@ -56,5 +56,5 @@
 // implementations over a long period of time.
 
 soroban_env_macros::generate_env_meta_consts!(
-    interface_version: 23,
+    interface_version: 24,
```
Comment (graydon) [inline soroban-env-common/src/meta.rs:59]: I don't think this needs an interface bump. Nothing in env.rs is changing

**Review by graydon — CHANGES_REQUESTED** [review]

File: soroban-env-common/src/meta.rs (hunk 1 above)
Comment (jayz22) [reply soroban-env-common/src/meta.rs]: I undid the version bump.

**Comment by jayz22** [comment]
@graydon I've undone the version bump. Also upgraded all the wasm dependencies of soroban from 0.0.4 to 0.0.6. Should be good to merge.

**Review by graydon — APPROVED** [review]

**Pull request merged**